librz/arch/isa/xtensa/xtensa_il.c

```diff
@@ -0,0 +1,2178 @@
+	return SEQ2(SETG(REGN(0), VARG("ps")),
+		PS_field_set(PS_INTLEVEL, U32(IMM(1))));
+}
+
+// FIXME: statusflags: VZI  + reciprocal_square_root_approximation
+static RzAnalysisLiftedILOp op_rsqrt0_s(XtensaContext *ctx) {
+	return NOP();
+}
+
+static RzAnalysisLiftedILOp op_rsr(XtensaContext *ctx) {
+	return SETG(REGN(0), IREG(1));
+}
+
+static RzAnalysisLiftedILOp op_rsync(XtensaContext *ctx) {
+	return NOP();
+}
+
+#define RUR_IMPL(S) \
+	static RzAnalysisLiftedILOp op_rur_##S(XtensaContext *ctx) { \
+		return SETG(REGN(0), VARG(#S)); \
+	}
+#define RUR_AE_IMPL(S) \
+	static RzAnalysisLiftedILOp op_rur_ae_##S(XtensaContext *ctx) { \
+		return SETG(REGN(0), VARG(#S)); \
+	}
+
+RUR_IMPL(accx_0);
+RUR_IMPL(accx_1);
+RUR_AE_IMPL(bithead);
+RUR_AE_IMPL(bitptr);
+RUR_AE_IMPL(bitsused);
+RUR_AE_IMPL(cbegin0);
+RUR_AE_IMPL(cend0);
+RUR_AE_IMPL(cwrap);
+RUR_AE_IMPL(cw_sd_no);
+RUR_AE_IMPL(first_ts);
+RUR_AE_IMPL(nextoffset);
+RUR_AE_IMPL(overflow);
+RUR_AE_IMPL(ovf_sar);
+RUR_AE_IMPL(sar);
+RUR_AE_IMPL(searchdone);
+RUR_AE_IMPL(tablesize);
+RUR_AE_IMPL(ts_fts_bu_bp);
+RUR_IMPL(fft_bit_width);
+RUR_IMPL(gpio_out);
+RUR_IMPL(qacc_h_0);
+RUR_IMPL(qacc_h_1);
+RUR_IMPL(qacc_h_2);
+RUR_IMPL(qacc_h_3);
+RUR_IMPL(qacc_h_4);
+RUR_IMPL(qacc_l_0);
+RUR_IMPL(qacc_l_1);
+RUR_IMPL(qacc_l_2);
+RUR_IMPL(qacc_l_3);
+RUR_IMPL(qacc_l_4);
+RUR_IMPL(sar_byte);
+RUR_IMPL(ua_state_0);
+RUR_IMPL(ua_state_1);
+RUR_IMPL(ua_state_2);
+RUR_IMPL(ua_state_3);
+
+static RzAnalysisLiftedILOp op_s16i(XtensaContext *ctx) {
+	return SEQ2(
+		SETL("vAddr", IMEM(1)),
+		STOREW(VARL("vAddr"), UNSIGNED(16, IREG(0))));
+}
+
+static RzAnalysisLiftedILOp op_s32c1i(XtensaContext *ctx) {
+	return SEQ4(
+		SETL("vAddr", IMEM(1)),
+		SETL("mem", LOADW(32, VARL("vAddr"))),
+		BRANCH(EQ(VARL("mem"), VARG("scompare1")),
+			STOREW(VARL("vAddr"), IREG(0)),
+			NOP()),
+		SETG(REGN(0), VARL("mem")));
+}
+
+// FIXME: ring
+static RzAnalysisLiftedILOp op_s32e(XtensaContext *ctx) {
+	return SEQ2(
+		SETL("vAddr", ADD(IREG(1), S32(IMM(2)))),
+		STOREW(VARL("vAddr"), IREG(0)));
+}
+
+static RzAnalysisLiftedILOp op_s32i(XtensaContext *ctx) {
+	return SEQ2(
+		SETL("vAddr", IMEM(1)),
+		STOREW(VARL("vAddr"), IREG(0)));
+}
+
+static RzAnalysisLiftedILOp op_s8i(XtensaContext *ctx) {
+	return SEQ2(
+		SETL("vAddr", IMEM(1)),
+		STORE(VARL("vAddr"), UNSIGNED(8, IREG(0))));
+}
+
+static RzAnalysisLiftedILOp op_sext(XtensaContext *ctx) {
+	return SETG(REGN(0), SEXTRACT32(IREG(1), U32(0), U32(IMM(2))));
+}
+
+static RzAnalysisLiftedILOp op_simcall(XtensaContext *ctx) {
+	return NOP();
+}
+
+static RzAnalysisLiftedILOp op_sll(XtensaContext *ctx) {
+	return SEQ2(
+		SETL("sa", EXTRACT32(VARG("sar"), U32(0), U32(5))),
+		SETG(REGN(0), SHIFTL0(IREG(1), VARL("sa"))));
+}
+
+static RzAnalysisLiftedILOp op_slli(XtensaContext *ctx) {
+	return SEQ2(
+		SETL("sa", U32(IMM(2))),
+		SETG(REGN(0), SHIFTL0(IREG(1), VARL("sa"))));
+}
+
+// TODO: see Divide and Square Root Sequences on page 110.
```

Comment:
Do you plan to fix this?

Reply:
I'm planning to do this in another new PR, and the instructions are so strange that I'm not sure it's necessary to implement them.

<img width="1065" alt="image" src="https://github.com/user-attachments/assets/f322f001-ef42-45d1-b6be-328f3459e354" />

Reply:
It depends if these instructions are used in the real code or not. Yep, it's okay to do in a subsequent PR.